extensions/mypy_extensions.py

```diff
@@ -7,16 +7,78 @@
 
 # NOTE: This module must support Python 2.7 in addition to Python 3.x
 
+import sys
+import types
+from typing import _type_check  # type: ignore
 
-def TypedDict(typename, fields):
-    """TypedDict creates a dictionary type that expects all of its
+
+def _check_fails(cls, other):
+    try:
+        if sys._getframe(1).f_globals['__name__'] not in ['abc', 'functools']:
+            raise TypeError('TypedDict does not support instance and class checks')
+    except (AttributeError, ValueError):
+        pass
+    return False
+
+def _dict_new(inst, cls, *args, **kwargs):
+    return dict(*args, **kwargs)
+
+def _typeddict_new(inst, cls, _typename, _fields=None, **kwargs):
+    if _fields is None:
+        _fields = kwargs
+    elif kwargs:
+        raise TypeError("TypedDict takes either a dict or "
+                        "keyword arguments, but not both")
```

Comment:
Can you break after the comma? (I worry about people grepping for the error message.)